silverback/application.py

```diff
@@ -142,6 +197,34 @@ def __get_user_taskdata_handler(self, task_type: TaskType) -> list[TaskData]:
     def __get_user_all_taskdata_handler(self) -> list[TaskData]:
         return [v for k, l in self.tasks.items() if str(k).startswith("user:") for v in l]
 
+    async def __load_snapshot_handler(self, startup_state: StateSnapshot):
+        # NOTE: *DO NOT USE* in Runner, as it will not be updated by the app
+        self.state = SharedState()
+        # NOTE: attribute does not exist before this task is executed,
+        #       ensuring no one uses it during worker startup
+
+        self.state["system:last_block_seen"] = startup_state.last_block_seen
+        self.state["system:last_block_processed"] = startup_state.last_block_processed
+        # TODO: Load user custom state (should not start with `system:`)
+
+    async def __create_snapshot_handler(
+        self,
+        last_block_seen: int | None = None,
+        last_block_processed: int | None = None,
+    ):
+        # Task that updates state checkpoints before/after every non-system runtime task/at shutdown
+        if last_block_seen is not None:
+            self.state["system:last_block_seen"] = last_block_seen
+
+        if last_block_processed is not None:
+            self.state["system:last_block_processed"] = last_block_processed
+
+        return StateSnapshot(
+            # TODO: Migrate these to parameters (remove explicitly from state)
+            last_block_seen=self.state.get("system:last_block_seen", -1),
+            last_block_processed=self.state.get("system:last_block_processed", -1),
+        )
```

Comment:
question: Do we have future plans for this handler? This more or less is just the runner sending these two values to the worker to send back to the runner.

Or maybe we intend to make this available to user code since it's called every checkpoint?

Reply:
State snapshots in general will be created using this mechanism (parameters will make use of this), but I added `system:` to the front of these specific internal state variables because then they cannot conflict with any attribute added to state directly by users (e.g. `app.state.something = ...` by definition cannot start with `system:`)

One of my ideas when adding parameters is that those are only allowed to be specific types, and only parameters will end up in state snapshots that get communicated back to the cluster services. All other non-parameter state will not be backed up in snapshots.

There may exist other features that make use of this, like nonce tracking